mpqp/noise/noise_model.py

```diff
@@ -178,7 +187,7 @@ def __init__(
                 )
 
         nb_targets = len(targets)
-        if nb_targets < dimension:
+        if nb_targets != 0 and nb_targets < dimension:
```

Comment:
The problem with this is that if targets==[], meaning we need all qubits of the circuit, we don't know how many exact qubit we will target at this point, so we cannot catch really an error with dimension.
Example, you have a 2-qubit circuit, you don't precise the target, so at runtime it will become [0,1], BUT, you have instantiated a noise model with dimension=3, here the problem will not be catched, meaning we need to do a double check somewhere else, or move this verification later in the execution ... ?

This requires for me some discussion

Reply:
It is solve in the branche remove target. But it will not comme in this release

Reply:
So in my sense we should remove the possibility to put target as optional and empty, and only include the whole working code when it will be the time for the right release

Reply:
yes agree

Reply:
agreed

Reply:
I like the solution proposed in 664b60d

Reply:
after more thought, this solution wasn't complete enough and thus was reverted in 971d97c